src/datachain/lib/tar.py

```diff
@@ -5,28 +5,23 @@
 from datachain.lib.file import File, TarVFile
 
 
-def build_tar_member(parent: File, info: tarfile.TarInfo) -> File:
+def build_tar_member(
+    parent: File, info: tarfile.TarInfo, file_cls: type[TarVFile] = TarVFile
+) -> TarVFile:
```

Comment:
should it be file_cls or substype of TarVFile?

Reply:
Can you clarify? It should be a an instance of a subtype of `TarVFile`. Is there a better way you see to specify the type here?

Reply:
yep, sorry, I think it goes into TypeVars, generics, or something. We want probably to bind type of the `file_cls` to the return type (to be the same). I'm not an expert here (I can take a look though), but probably it's achieved via TypeVars.

something like:

```python
T = TypeVar('T', bound=TarVFile)
def build_tar_member(parent: File, info: tarfile.TarInfo, file_cls: type[T] = TarVFile
) -> T:
   pass
```

It seems not possible though in mypy? https://github.com/python/mypy/issues/9003 ... feel free to disregard this.

Reply:
Got it. That makes sense, but it does appear it's not supported well enough in mypy. Your syntax above LGTM but fails to pass the mypy check, which doesn't seem to acknowledge that `TarVFile` is of type `T`.